opm/simulators/wells/StandardWellV_impl.hpp

```diff
@@ -2265,12 +2273,42 @@ namespace Opm
     void
     StandardWellV<TypeTag>::
     computeWellRatesWithBhp(const Simulator& ebosSimulator,
-                            const EvalWell& bhp,
+                            const std::vector<Scalar>& B_avg,
+                            const double& bhp,
+                            const bool iterate,
                             std::vector<double>& well_flux,
-                            Opm::DeferredLogger& deferred_logger) const
+                            Opm::DeferredLogger& deferred_logger)
     {
+
         const int np = number_of_phases_;
         well_flux.resize(np, 0.0);
+        WellControls* wc = well_controls_;
```

Comment:
Without `iterate` to be `true`,  the setting of the `bhp` limit should not happen here.  An immediate change will be put these into the `if (iterate) {`, including the following code
```
        // reset bhp limit
        well_controls_iset_target(wc, bhp_index, orig_bhp);
        well_controls_set_current(wc, orig_current);
    }	

```
Please do not do anything now to see whether we can find a better solution.